ts/src/xt.ts

```diff
@@ -0,0 +1,4094 @@
+        //                     "side": "SUB",
+        //                     "afterAmount": null,
+        //                     "createdTime": 1679116769914
+        //                 },
+        //             ]
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'result', {});
+        const ledger = this.safeValue (data, 'items', []);
+        return this.parseLedger (ledger, currency, since, limit);
+    }
+
+    parseLedgerEntry (item, currency = undefined) {
+        //
+        //     {
+        //         "id": "207260567109387524",
+        //         "coin": "usdt",
+        //         "symbol": "btc_usdt",
+        //         "type": "FEE",
+        //         "amount": "-0.0213",
+        //         "side": "SUB",
+        //         "afterAmount": null,
+        //         "createdTime": 1679116769914
+        //     }
+        //
+        const side = this.safeString (item, 'side');
+        const direction = (side === 'ADD') ? 'in' : 'out';
+        const currencyId = this.safeString (item, 'coin');
+        const timestamp = this.safeInteger (item, 'createdTime');
+        return {
+            'id': this.safeString (item, 'id'),
+            'direction': direction,
+            'account': undefined,
+            'referenceId': undefined,
+            'referenceAccount': undefined,
+            'type': this.parseLedgerEntryType (this.safeString (item, 'type')),
+            'currency': this.safeCurrencyCode (currencyId, currency),
+            'amount': this.safeNumber (item, 'amount'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'before': undefined,
+            'after': this.safeNumber (item, 'afterAmount'),
+            'status': undefined,
+            'fee': {
+                'currency': undefined,
+                'cost': undefined,
+            },
+            'info': item,
+        };
+    }
+
+    parseLedgerEntryType (type) {
+        const ledgerType = {
+            'EXCHANGE': 'transfer',
+            'CLOSE_POSITION': 'trade',
+            'TAKE_OVER': 'trade',
+            'MERGE': 'trade',
+            'QIANG_PING_MANAGER': 'fee',
+            'FUND': 'fee',
+            'FEE': 'fee',
+            'ADL': 'auto-deleveraging',
+        };
+        return this.safeString (ledgerType, type, type);
+    }
+
+    async fetchDepositAddress (code: string, params = {}) {
+        /**
+         * @method
+         * @name xt#fetchDepositAddress
+         * @description fetch the deposit address for a currency associated with this account
+         * @see https://doc.xt.com/#deposit_withdrawaldepositAddressGet
+         * @param {string} code unified currency code
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {object} an [address structure]{@link https://docs.ccxt.com/en/latest/manual.html#address-structure}
+         */
+        await this.loadMarkets ();
```

Comment:
@Dan-krm getting a weird response here, 
```
p xt fetchDepositAddress "USDT"
Python v3.10.9
CCXT v3.0.51
xt.fetchDepositAddress(USDT)
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt2/ccxt/examples/py/cli.py", line 232, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/cjg/Git/ccxt2/ccxt/examples/py/cli.py", line 211, in main
    result = await method(*args)
  File "/Users/cjg/Git/ccxt2/ccxt/python/ccxt/async_support/xt.py", line 2838, in fetch_deposit_address
    response = await self.privateSpotGetDepositAddress(self.extend(request, params))
  File "/Users/cjg/Git/ccxt2/ccxt/python/ccxt/async_support/base/exchange.py", line 1613, in request
    return await self.fetch2(path, api, method, params, headers, body, config, context)
  File "/Users/cjg/Git/ccxt2/ccxt/python/ccxt/async_support/base/exchange.py", line 1610, in fetch2
    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File "/Users/cjg/Git/ccxt2/ccxt/python/ccxt/async_support/base/exchange.py", line 180, in fetch
    self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File "/Users/cjg/Git/ccxt2/ccxt/python/ccxt/async_support/xt.py", line 3791, in handle_errors
    raise ExchangeError(feedback)
ccxt.base.errors.ExchangeError: xt {"rc":1,"mc":"FAILURE","ma":[],"result":null}
xt requires to release all resources with an explicit call to the .close() coroutine. If you are using the exchange instance with async coroutines, add `await exchange.close()` to your code into a place when you're done with the exchange and don't need the exchange instance anymore (at the end of your async coroutine).
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x1045a6140>
```

Reply:
fetchDepositAddress requires a chain parameter on xt, and for USDT there's no default, should I set a default for USDT or maybe throw an error if networkId isn't defined?

Reply:
> fetchDepositAddress requires a chain parameter on xt, and for USDT there's no default, should I set a default for USDT or maybe throw an error if networkId isn't defined?

yeah I think you can require it